lib/plugins/paragon-webpack-plugin/ParagonWebpackPlugin.js

```diff
@@ -0,0 +1,230 @@
+const { Compilation, sources } = require('webpack');
+const parse5 = require('parse5');
+const {
+  getParagonVersion,
+  getParagonThemeCss,
+  replacePeriodsWithHyphens,
+} = require('../../../config/data/paragonUtils');
+
+const paragonVersion = getParagonVersion(process.cwd());
+const paragonThemeCss = getParagonThemeCss(process.cwd());
+
+const brandVersion = getParagonVersion(process.cwd(), { isBrandOverride: true });
+const brandThemeCss = getParagonThemeCss(process.cwd(), { isBrandOverride: true });
+
+class ParagonWebpackPlugin {
+  constructor() {
+    this.paragon = {
+      version: paragonVersion,
+      coreEntryName: undefined,
+      themeVariantEntryNames: {},
+    };
+
+    this.brand = {
+      version: brandVersion,
+      coreEntryName: undefined,
+      themeVariantEntryNames: {},
+    };
+
+    if (!paragonThemeCss && !brandThemeCss) {
+      return;
+    }
+
+    if (paragonThemeCss) {
+      // Core Paragon
+      this.paragon.coreEntryName = replacePeriodsWithHyphens(paragonThemeCss.core.entryName);
+      Object.entries(paragonThemeCss.variants).forEach(([key, value]) => {
+        this.paragon.themeVariantEntryNames[key] = {
+          entryName: replacePeriodsWithHyphens(value.entryName),
+          default: value.default,
+          dark: value.dark,
+        };
+      });
+    }
+
+    if (brandThemeCss) {
+      // `@edx/brand` overrides
+      this.brand.coreEntryName = replacePeriodsWithHyphens(brandThemeCss.core.entryName);
+      Object.entries(brandThemeCss.variants).forEach(([key, value]) => {
+        this.brand.themeVariantEntryNames[key] = {
+          entryName: replacePeriodsWithHyphens(value.entryName),
+          default: value.default,
+          dark: value.dark,
+        };
+      });
+    }
+  }
+
+  logger(message) {
```

Comment:
I don't think this method is used anywhere and also it probably won't work because `console` is a browser API and webpack does not have access to it.

Reply:
Correct, `logger` is no longer used anywhere (was helpful to have around for debugging during development). FWIW, though, the `console.log` works during `npm run build` and `npm run start`, running in the terminal through Node.js.

Reply:
Yeah, I think I had a brain lag moment when I wrote this comment about `console.log`, I've used it the same way thousand times 🤦 It does exist in Node.js.

Reply:
I'm likely going to keep it around due to its helpfulness for debugging/adding temporary logs haha. Just used it to debug a few things to respond to your other comments 😅

Reply:
Now that we confirmed that it works, I don't mind 😅